src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java

```diff
@@ -92,6 +95,31 @@ public void shouldThrowCBSApplicationExceptionWhenBookingAlreadyApprovedOrNotInA
 		
 		bookingHrApproveHelper.hrApprove(booking);
 	}
+	
+	@Test
+	public void shouldThrowExceptionWhileGeneratingPDF_HRApprove() {
+		
+		Booking booking = createBooking();
+		BookingRevision bookingRevision = createBookingRevision();
+		CBSUser cbsUser = createCBSUser();
+		
+		when(loggedInUserService.isCurrentUserInHRRole()).thenReturn(true);
+		when(bookingSaveHelper.getLatestRevision(booking)).thenReturn(bookingRevision);
+		when(html2PdfService.generateAgreementPdf(bookingRevision)).thenThrow(IllegalStateException.class);
+		when(loggedInUserService.getLoggedInUserDetails()).thenReturn(cbsUser);
+		when(bookingSaveHelper.saveBooking(booking, bookingRevision, 1006L,cbsUser)).thenReturn(booking);
+		
+		bookingHrApproveHelper.hrApprove(booking);
+		
+		verify(loggedInUserService).isCurrentUserInHRRole();
```

Comment:
break the verify in separate test cases.

Reply:
as per your suggestion. I have implemented in all new Test classes but this an old file, so kept as it is

Reply:
yes Akshay also told me ... i asked if you can then do it. he is implementing ... check on your end if you can do these changes

Reply:
ok.. i am doing changes